sound/soc/sof/intel/pci-ptl.c

```diff
@@ -0,0 +1,77 @@
+// SPDX-License-Identifier: (GPL-2.0-only OR BSD-3-Clause)
+//
+// This file is provided under a dual BSD/GPLv2 license.  When using or
+// redistributing this file, you may do so under either license.
+//
+// Copyright(c) 2024 Intel Corporation.
+//
+
+#include <linux/module.h>
+#include <linux/pci.h>
+#include <sound/soc-acpi.h>
+#include <sound/soc-acpi-intel-match.h>
+#include <sound/sof.h>
+#include "../ops.h"
+#include "../sof-pci-dev.h"
+
+/* platform specific devices */
+#include "hda.h"
+#include "mtl.h"
+
+static const struct sof_dev_desc ptl_desc = {
+	.use_acpi_target_states	= true,
+	.machines               = snd_soc_acpi_intel_ptl_machines,
+	.alt_machines		= snd_soc_acpi_intel_ptl_sdw_machines,
+	.resindex_lpe_base      = 0,
+	.resindex_pcicfg_base   = -1,
+	.resindex_imr_base      = -1,
+	.irqindex_host_ipc      = -1,
+	.chip_info		= &ptl_chip_info,
+	.ipc_supported_mask	= BIT(SOF_IPC_TYPE_4),
+	.ipc_default		= SOF_IPC_TYPE_4,
+	.dspless_mode_supported	= true,
+	.default_fw_path = {
+		[SOF_IPC_TYPE_4] = "intel/sof-ipc4/ptl",
+	},
+	.default_lib_path = {
+		[SOF_IPC_TYPE_4] = "intel/sof-ipc4-lib/ptl",
+	},
+	.default_tplg_path = {
+		[SOF_IPC_TYPE_4] = "intel/sof-ipc4-tplg",
+	},
+	.default_fw_filename = {
+		[SOF_IPC_TYPE_4] = "sof-ptl.ri",
```

Comment:
This will pollute everyone's kernel logs with a "firmware not found" error, same as `sof-lnl.ri` right now:

https://intel-gfx-ci.01.org/tree/intel-xe/xe-1239-5a43da669cdb9b8df66e32a661b09cd9c52e35f2/shard-lnl-1/boot21.txt


```
<6>[   26.307550] sof-audio-pci-intel-lnl 0000:00:1f.3:  Firmware file: intel/sof-ipc4/lnl/sof-lnl.ri
<6>[   26.307551] sof-audio-pci-intel-lnl 0000:00:1f.3:  Topology file: intel/sof-ipc4-tplg/sof-hda-generic-idisp-4ch.tplg
<6>[   26.307553] sof-audio-pci-intel-lnl 0000:00:1f.3: Check if you have 'sof-firmware' package installed.
<6>[   26.307554] sof-audio-pci-intel-lnl 0000:00:1f.3: Optionally it can be manually downloaded from:
<6>[   26.307555] sof-audio-pci-intel-lnl 0000:00:1f.3:    https://github.com/thesofproject/sof-bin/
<4>[   26.309505] sof-audio-pci-intel-lnl 0000:00:1f.3: Direct firmware load for (null)/(null) failed with error -2
<3>[   26.309527] sof-audio-pci-intel-lnl 0000:00:1f.3: error: sof firmware file is missing, you might need to
<3>[   26.309528] sof-audio-pci-intel-lnl 0000:00:1f.3:        download it from https://github.com/thesofproject/sof-bin/
```
Also, the error message will mislead people to https://github.com/thesofproject/sof-bin/ which will not have any sof-ptl.ri until PTL is officially signed.

As long as we don't release any "development" sof-ptl.ri, everyone working with PTL should "temporarily" blacklist `sof-audio-pci-intel-ptl` in their CI - and of course forget about it and never enable it again.

cc: @lucasdemarchi
- https://github.com/thesofproject/kconfig/pull/91#issuecomment-2080241578

Reply:
I don't see any reason to change or disable this error. 

By the time platforms are available commercially from OEMs, there will be an sof-bin release with the relevant firmware signed.

Someone involved in pre-production releases should know where to get the firmware.

Reply:
> Someone involved in pre-production releases...

In practice there is no such thing. So I will just guide the (Intel) people who asked me how to fix this error to `blocklist` SOF.

Same as we do:
- https://github.com/thesofproject/sof-test/issues/1198